pmlc/dialect/eltwise/ir/predicates.td

```diff
@@ -31,7 +31,7 @@ def ScalarBool : Type<And<[
 
 def ScalarIndex : Type<And<[
     IsScalarPred,
-    CPred<"$_self.cast<ScalarType>().type() == DataType::i32">
+    CPred<"$_self.cast<ScalarType>().type() == DataType::i32 || $_self.cast<ScalarType>().type() == DataType::i64">
```

Comment:
We have been considering adding a new `DataType` enum value to represent machine-word sized integers (e.g. indexes). I'm OK with doing it like this for now, but I expect there to be some churn on this as we start getting lowering working on different targets.